core/arch/x86/decode.c

```diff
@@ -1720,6 +1734,17 @@ decode_modrm(decode_info_t *di, byte optype, opnd_size_t opsize, opnd_t *reg_opn
              * specify a segment selector and address.  The opcode must be
              * examined to know how to interpret those 6 bytes.
              */
+            if (di->evex_encoded) {
+                if (di->mod == 1) {
+                    int compressed_disp_scale = instr_get_compressed_disp_scale(
+                        di->tuple_type, TEST(PREFIX_EVEX_b, di->prefixes),
+                        instr_get_input_size_from_opcode(
+                            di->opcode, TEST(di->prefixes, PREFIX_REX_W)),
+                        instr_get_vector_length(TEST(di->prefixes, PREFIX_VEX_L),
+                                                TEST(di->prefixes, PREFIX_EVEX_LL)));
```

Comment:
Rather than duplicating all of these parameters multiple times, how about having the *_get_compressed_disp_scale() just taking in `di` and nothing else.

Reply:
Ok, done.